src/dfx/src/lib/canister_logs/log_visibility.rs

```diff
@@ -0,0 +1,129 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::util::clap::parsers::log_visibility_parser;
+use candid::Principal;
+use clap::{ArgAction, Args};
+use ic_utils::interfaces::management_canister::{LogVisibility, StatusCallResult};
+
+#[derive(Args, Clone, Debug, Default)]
+pub struct LogVisibilityOpt {
+    /// Specifies who is allowed to read the canister's logs.
+    /// Can be either "controllers" or "public".
+    #[arg(
+        long,
+        value_parser = log_visibility_parser,
+        conflicts_with("add_log_viewer"),
+        conflicts_with("remove_log_viewer"),
+        conflicts_with("set_log_viewer"),
+    )]
+    log_visibility: Option<LogVisibility>,
+
+    /// Add a principal to the list of log viewers of the canister.
+    #[arg(long, action = ArgAction::Append, conflicts_with("set_log_viewer"))]
+    add_log_viewer: Option<Vec<String>>,
```

Comment:
Doc text, the field name and the field type do not match. Doc text and field name suppose it only adds a single principal, while the type expects a vector of strings (which I assume is a vector of principals).

I'd expect it either to be plural or to be an optional string. Let me know if my assumptions are incorrect.

Reply:
You can add multiple viewers in one command like below

Reply:
This is why the action is `ArgAction::Append`.